doc/source/cluster/kubernetes/k8s-ecosystem/volcano.md

```diff
@@ -0,0 +1,319 @@
+  - This label only works after the creation of a `PriorityClass` resource
+  - ```shell
+    labels:
+      ray.io/scheduler-name: volcano
+      ray.io/priority-class-name: <replace with correct PriorityClass resource name>
+    ```
+- `volcano.sh/queue-name`: the Volcano [queue](https://volcano.sh/en/docs/queue/) name the cluster will be submitted to.
+  - This label only works after the creation of a `Queue` resource
+  - ```shell
+    labels:
+      ray.io/scheduler-name: volcano
+      volcano.sh/queue-name: <replace with correct Queue resource name>
+    ```
+
+If autoscaling is enabled, `minReplicas` will be used for gang scheduling, otherwise the desired `replicas` will be used.
+
+### Step 5: Use Volcano for batch scheduling
+
+If you need some guidance, check out [examples](https://github.com/volcano-sh/volcano/tree/master/example) available.
+
+## Example
+
+Before going through the example, remove any ray clusters running to ensure successful run through of the example below. 
+```shell
+kubectl delete raycluster --all
+```
+
+### Gang scheduling
+
+In this example, we'll walk through how gang scheduling works with Volcano and KubeRay.
+
+First, let's create a queue with a capacity of 4 CPUs and 6Gi of RAM:
+
+```shell
+kubectl create -f - <<EOF
+apiVersion: scheduling.volcano.sh/v1beta1
+kind: Queue
+metadata:
+  name: kuberay-test-queue
+spec:
+  weight: 1
+  capability:
+    cpu: 4
+    memory: 6Gi
+EOF
+```
+
+The **weight** in the definition above indicates the relative weight of a queue in cluster resource division. This is useful in cases where the total **capability** of all the queues in your cluster exceeds the total available resources, forcing the queues to share among themselves. Queues with higher weight will be allocated a proportionally larger share of the total resources.
+
+The **capability** is a hard constraint on the maximum resources the queue will support at any given time. It can be updated as needed to allow more or fewer workloads to run at a time.
+
+Next we'll create a RayCluster with a head node (1 CPU + 2Gi of RAM) and two workers (1 CPU + 1Gi of RAM each), for a total of 3 CPU and 4Gi of RAM:
+
+```shell
+# Path: kuberay/ray-operator/config/samples
+# Includes labels `ray.io/scheduler-name: volcano` and `volcano.sh/queue-name: kuberay-test-queue` in the metadata.labels
+curl -LO https://raw.githubusercontent.com/ray-project/kuberay/master/ray-operator/config/samples/ray-cluster.volcano-scheduler-queue.yaml
+kubectl apply -f ray-cluster.volcano-scheduler-queue.yaml
+```
+
+Because our queue has a capacity of 4 CPU and 6Gi of RAM, this resource should schedule successfully without any issues. We can verify this by checking the status of our cluster's Volcano PodGroup to see that the phase is `Running` and the last status is `Scheduled`:
```

Comment:
```suggestion
Because the queue has a capacity of 4 CPU and 6Gi of RAM, this resource should schedule successfully without any issues. You can verify this by checking the status of the cluster's Volcano PodGroup to see that the phase is `Running` and the last status is `Scheduled`:
```